### PR TITLE
Fix normal_() reproducibility for non-contiguous tensors on CPU

### LINKED ISSUE #166856: CPU `normal_()` Non-Deterministic Behavior with Non-Contiguous Tensors
### 🐛 Describe the bug

The `torch.Tensor.normal_()` in-place operation produces **different random values** for contiguous vs non-contiguous tensors on CPU, even when using the same seed. This violates reproducibility expectations and creates platform-dependent behavior, as CUDA correctly produces identical results for both memory layouts.

**Expected behavior:** With the same seed, `normal_()` should produce identical random values regardless of tensor contiguity (both CPU and CUDA should behave this way).

**Actual behavior:** On CPU, `normal_()` produces completely different values (max difference ~3.85) for contiguous vs non-contiguous tensors. CUDA correctly produces identical values (max difference 0.0).

### Minimal reproduction code

```python
import torch

# Test on CPU
torch.manual_seed(42)
x_contiguous = torch.zeros(10, 20)
x_contiguous.normal_(mean=0, std=1)

torch.manual_seed(42)  # Reset to same seed
x_noncontiguous = torch.zeros(10, 20).T.clone()  # (20, 10) non-contiguous
assert not x_noncontiguous.is_contiguous()
x_noncontiguous.normal_(mean=0, std=1)

# These should be identical (accounting for transpose)
print("Values match:", torch.allclose(x_contiguous.T, x_noncontiguous))
# Expected: True
# Actual on CPU: False ❌
# Actual on CUDA: True ✅

# Show the difference
print("Max difference:", (x_contiguous.T - x_noncontiguous).abs().max().item())
# On CPU: ~3.85 (completely different values!)
# On CUDA: ~0.0 (correctly identical)
```

[reproduce_cpu_normal_bug.py](https://github.com/user-attachments/files/23303865/reproduce_cpu_normal_bug.py)

### Versions

Collecting environment information...
PyTorch version: 2.9.0+cu128
Is debug build: False
CUDA used to build PyTorch: 12.8
ROCM used to build PyTorch: N/A

OS: Ubuntu 24.04.2 LTS (x86_64)
GCC version: (Ubuntu 13.3.0-6ubuntu2~24.04) 13.3.0
Clang version: 15.0.0 (git@github.com:llvm/llvm-project.git 4ba6a9c9f65bbc8bd06e3652cb20fd4dfc846137)
CMake version: version 3.28.3
Libc version: glibc-2.39

Python version: 3.12.3 (main, Aug 14 2025, 17:47:21) [GCC 13.3.0] (64-bit runtime)
Python platform: Linux-6.14.0-27-generic-x86_64-with-glibc2.39
Is CUDA available: True
CUDA runtime version: Could not collect
CUDA_MODULE_LOADING set to:
GPU models and configuration: GPU 0: NVIDIA GeForce RTX 4070
Nvidia driver version: 575.57.08
cuDNN version: Probably one of the following:
/usr/lib/x86_64-linux-gnu/libcudnn.so.8.9.7
/usr/lib/x86_64-linux-gnu/libcudnn_adv_infer.so.8.9.7
/usr/lib/x86_64-linux-gnu/libcudnn_adv_train.so.8.9.7
/usr/lib/x86_64-linux-gnu/libcudnn_cnn_infer.so.8.9.7
/usr/lib/x86_64-linux-gnu/libcudnn_cnn_train.so.8.9.7
/usr/lib/x86_64-linux-gnu/libcudnn_ops_infer.so.8.9.7
/usr/lib/x86_64-linux-gnu/libcudnn_ops_train.so.8.9.7
Is XPU available: False
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

CPU:
Architecture:                         x86_64
CPU op-mode(s):                       32-bit, 64-bit
Address sizes:                        39 bits physical, 48 bits virtual
Byte Order:                           Little Endian
CPU(s):                               24
On-line CPU(s) list:                  0-23
Vendor ID:                            GenuineIntel
Model name:                           13th Gen Intel(R) Core(TM) i7-13700F
CPU family:                           6
Model:                                183
Thread(s) per core:                   2
Core(s) per socket:                   16
Socket(s):                            1
Stepping:                             1
CPU(s) scaling MHz:                   33%
CPU max MHz:                          5200.0000
CPU min MHz:                          800.0000
BogoMIPS:                             4224.00
Flags:                                fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf tsc_known_freq pni pclmulqdq dtes64 monitor ds_cpl vmx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb ssbd ibrs ibpb stibp ibrs_enhanced tpr_shadow flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid rdseed adx smap clflushopt clwb intel_pt sha_ni xsaveopt xsavec xgetbv1 xsaves split_lock_detect user_shstk avx_vnni dtherm ida arat pln pts hwp hwp_notify hwp_act_window hwp_epp hwp_pkg_req hfi vnmi umip pku ospke waitpkg gfni vaes vpclmulqdq rdpid movdiri movdir64b fsrm md_clear serialize arch_lbr ibt flush_l1d arch_capabilities
Virtualization:                       VT-x
L1d cache:                            640 KiB (16 instances)
L1i cache:                            768 KiB (16 instances)
L2 cache:                             24 MiB (10 instances)
L3 cache:                             30 MiB (1 instance)
NUMA node(s):                         1
NUMA node0 CPU(s):                    0-23
Vulnerability Gather data sampling:   Not affected
Vulnerability Ghostwrite:             Not affected
Vulnerability Itlb multihit:          Not affected
Vulnerability L1tf:                   Not affected
Vulnerability Mds:                    Not affected
Vulnerability Meltdown:               Not affected
Vulnerability Mmio stale data:        Not affected
Vulnerability Reg file data sampling: Mitigation; Clear Register File
Vulnerability Retbleed:               Not affected
Vulnerability Spec rstack overflow:   Not affected
Vulnerability Spec store bypass:      Mitigation; Speculative Store Bypass disabled via prctl
Vulnerability Spectre v1:             Mitigation; usercopy/swapgs barriers and __user pointer sanitization
Vulnerability Spectre v2:             Mitigation; Enhanced / Automatic IBRS; IBPB conditional; PBRSB-eIBRS SW sequence; BHI BHI_DIS_S
Vulnerability Srbds:                  Not affected
Vulnerability Tsx async abort:        Not affected

Versions of relevant libraries:
[pip3] mypy_extensions==1.1.0
[pip3] numpy==2.3.4
[pip3] nvidia-cublas-cu12==12.8.4.1
[pip3] nvidia-cuda-cupti-cu12==12.8.90
[pip3] nvidia-cuda-nvrtc-cu12==12.8.93
[pip3] nvidia-cuda-runtime-cu12==12.8.90
[pip3] nvidia-cudnn-cu12==9.10.2.21
[pip3] nvidia-cufft-cu12==11.3.3.83
[pip3] nvidia-curand-cu12==10.3.9.90
[pip3] nvidia-cusolver-cu12==11.7.3.90
[pip3] nvidia-cusparse-cu12==12.5.8.93
[pip3] nvidia-cusparselt-cu12==0.7.1
[pip3] nvidia-nccl-cu12==2.27.5
[pip3] nvidia-nvjitlink-cu12==12.8.93
[pip3] nvidia-nvtx-cu12==12.8.90
[pip3] torch==2.9.0
[pip3] torchmonarch==0.1.2
[pip3] torchshow==0.5.2
[pip3] torchx-nightly==2025.11.2
[pip3] triton==3.5.0
[conda] Could not collect

cc @pbelevich

### PR DESCRIPTION
**Pull request by ahkush**
Fixes #166856 

**Problem**

The CPU implementation of ```normal_()``` used different code paths for contiguous vs non-contiguous tensors. Contiguous tensors used an optimized batched Box-Muller transform with ```(j, j+8)``` pairing in raw memory order, while non-contiguous tensors used ```TensorIterator``` with consecutive ```(j, j+1)``` pairing in stride order. This caused identical seeds to produce completely different random values depending on tensor memory layout, breaking reproducibility. CUDA did not have this issue as it used a single unified path for both cases.

**Solution**

Removed the specialized contiguous-only optimizations ```(normal_fill_AVX2, normal_fill_VSX, normal_fill, normal_fill_16)``` and unified all CPU normal distribution sampling to use ```TensorIterator with at::normal_distribution```, matching CUDA's behavior. This ensures ```normal_()``` produces identical results for the same seed regardless of tensor contiguity or memory layout.

cc @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10 @jerryzh168 @aditew01

### CODE REVIEW

**Comment by ahkush** [comment]
cc: @vishalgoyal316 @cleonard530 @gaurav-redhat

**Comment by janeyx99** [comment]
This sounds like it'd lead to a performance drop in the most common case for contiguous tensors. Please measure what that is in a benchmark and share the results.
